client/src/featureform/serving.py

```diff
@@ -257,11 +257,12 @@ def feature_df_from_csv(self, feature, filename):
 
     def merge_feature_into_ts(self, feature_row, label_row, df, trainingset_df):
         if feature_row['source_timestamp'] != "":
-            trainingset_df = pd.merge_asof(trainingset_df, df.sort_values(['ts']), direction='backward',
+            trainingset_df = pd.merge_asof(trainingset_df, df.sort_values(feature_row['source_timestamp']), direction='backward',
                                            left_on=label_row['source_timestamp'],
                                            right_on=feature_row['source_timestamp'], left_by=label_row['source_entity'],
                                            right_by=feature_row['source_entity'])
-            df.drop_duplicates(subset=[feature_row['source_entity']], keep="last", inplace=True)
+            if feature_row['source_timestamp'] != label_row['source_timestamp']:
```

Comment:
so this drops a feature's timestamp column if they do not match the label's timestamp? Is this because only the label timestamp is relevant?

Reply:
It iterates through all the features and does point in time matching with each one. Sometimes they have the same timestamp column name, so we don't want to drop since it'll drop the labels timestamp. But if it has a different timestamp we drop it because its not useful anymore

Reply:
ok makes sense 👍